samples/unit_testing/subscription-manager/Auth/authorization.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import azure.functions as func
+from .usertoken import UserToken
+from functools import wraps
+
+"""
+Decorator method that is called for authorization before the Durable HTTP start method is invoked.
+It uses the X-MS-CLIENT-PRINCIPAL id in the request header to authenticate against a set of known
+Subscription Managers/Readers
+"""
+def authorize(allowed_groups:list):
+    """Wrap the decorator to allow passing in a group name"""
+    def decorator_authorize(decorated_function):
+        """Decorator to handle authorization"""
+
+        # Wraps ensures that the decorated function's parameters are exposed and not
+        # this function's parameters.
+        @wraps(decorated_function)
```

Comment:
My understanding was that `wraps` is primarily used to preserve the *metadata* of a function (such as docstrings and attributes like `__name__`) and not necessarily for preserving parameters. Can you please clarify how/why it's being used here?